sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -237,17 +237,20 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
 }
 
 object PROCESS_TABLES extends QueryTest with SQLTestUtils {
-  val releaseMirror = "https://dist.apache.org/repos/dist/release"
+  val releaseMirror = sys.env.getOrElse("SPARK_RELEASE_MIRROR",
+    "https://dist.apache.org/repos/dist/release")
   // Tests the latest version of every release line.
   val testingVersions: Seq[String] = {
     import scala.io.Source
     val versions: Seq[String] = try {
-      Source.fromURL(s"${releaseMirror}/spark").mkString
-        .split("\n")
-        .filter(_.contains("""<li><a href="spark-"""))
-        .filterNot(_.contains("preview"))
-        .map("""<a href="spark-(\d.\d.\d)/">""".r.findFirstMatchIn(_).get.group(1))
-        .filter(_ < org.apache.spark.SPARK_VERSION)
+      Utils.tryWithResource(Source.fromURL(s"$releaseMirror/spark")) { htmlContent =>
+        htmlContent.mkString
+          .split("\n")
+          .filter(_.contains("""<a href="spark-"""))
```

Comment:
remove `<li>` because not all pages have this tag, like [https://mirror.bit.edu.cn/apache/spark/](view-source:https://mirror.bit.edu.cn/apache/spark/)

Reply:
Is this safe for the official mirror?

Reply:
![image](https://user-images.githubusercontent.com/1475305/105565397-8cc71400-5d61-11eb-98c0-56e53695bcc3.png)

Yes, it looks like the result is the same

Reply:
@dongjoon-hyun  I'm not sure if the image can be seen, the verification process is as follows:
```
scala> val releaseMirror = "https://dist.apache.org/repos/dist/release"
val releaseMirror: String = https://dist.apache.org/repos/dist/release

scala> scala.io.Source.fromURL(s"${releaseMirror}/spark").mkString
     |         .split("\n")
     |         .filter(_.contains("""<a href="spark-"""))
     |         .filterNot(_.contains("preview"))
     |         .map("""<a href="spark-(\d.\d.\d)/">""".r.findFirstMatchIn(_).get.group(1))
val res0: Array[String] = Array(2.4.7, 3.0.1)

scala> scala.io.Source.fromURL(s"${releaseMirror}/spark").mkString
     |         .split("\n")
     |         .filter(_.contains("""<li><a href="spark-"""))
     |         .filterNot(_.contains("preview"))
     |         .map("""<a href="spark-(\d.\d.\d)/">""".r.findFirstMatchIn(_).get.group(1))
val res1: Array[String] = Array(2.4.7, 3.0.1)

```

Reply:
Got it. Thank you for the detail, @LuciferYang . I'll take a look at this more.

Reply:
Ok ~